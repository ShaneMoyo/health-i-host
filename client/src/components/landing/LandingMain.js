import React from 'react';

export default function LandingMain() {
  return <section class="animated fadeIn hero is-info is-fullheight">
          <div class="hero-body">
            <div class="column is-6 is-offset-3 has-text-centered">
              <div class="tile is-parent">
                <article class="tile is-child notification is-success landing">
                  <div class="content">
                    <p class="title">Healthihost</p>
                    <p class="subtitle">Enabling the Vitality of the body Naturally</p>
                  </div>
                </article>
              </div>
            </div>
          </div>
        </section>

}
