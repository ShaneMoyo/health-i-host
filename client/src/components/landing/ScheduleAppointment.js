import React from 'react';
import { NavLink } from 'react-router-dom';

const NavBarLink = props => <NavLink {...props} className="nav-link tile is-parent" activeClassName="active"/>;

export default function ScheduleAppointment() {
  return <div class="container has-text-centered">
          <div class="subtitle has-text-centered">
            A healthy body allows us to live an optimal life with energy and enthusiasm.
            Do you want to be in balance, free from pain, able to be active, and doing your heartâ€™s desires?
          </div>
          <br/>
          <div class="subtitle has-text-centered">
            <NavBarLink exact to="/appointment">
              <div class="tile is-parent is-vertical grow more">
                <article class="tile is-child notification has-text-white is-info">
                <p class="title">
                  Schedule Appointment
                </p>
                </article>
              </div>
            </NavBarLink>
          </div>
        </div>

}
