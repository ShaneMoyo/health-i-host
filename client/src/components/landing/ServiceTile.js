import React from 'react';
import { NavLink } from 'react-router-dom';
import Icon from '@mdi/react';

const NavBarLink = props => <NavLink {...props} className="nav-link tile is-parent" activeClassName="active"/>;

export default function ServiceTile(props) {
  const { icon, color, link, title } = props;
  const className = `tile is-child notification has-text-white is-${color} grow`;

  return <NavBarLink exact to={link}>
          <article class={className}>
            <span class="icon is-large is-pulled-left" >
              <Icon path={icon}
              size={2}
              color="white"
              />
            </span>
            <p class="title">{title}</p>
            <br/>
            <p class="subtitle">Learn More</p>
          </article>
        </NavBarLink>;

}
