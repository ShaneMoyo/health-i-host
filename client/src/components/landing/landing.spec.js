import React from 'react';
import { shallow } from 'enzyme';
import Landing from './landing';
import LandingMain from './LandingMain';
import ScheduleAppointment from './ScheduleAppointment';
import ServiceTile from './ServiceTile';


describe('Landing', () => {
  it('should render correctly Landing component', () => {
    const component = shallow(<Landing />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly LandingMain component', () => {
    const component = shallow(<LandingMain />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly LandingMain component', () => {
    const component = shallow(<LandingMain />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly ScheduleAppointment component', () => {
    const component = shallow(<ScheduleAppointment />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly ServiceTile component', () => {
    const component = shallow(<ServiceTile />);

    expect(component).toMatchSnapshot();
  });

});
