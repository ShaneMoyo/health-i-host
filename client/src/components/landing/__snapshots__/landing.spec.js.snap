// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should render correctly Landing component 1`] = `
<body>
  <LandingMain />
  <div
    class="box is-white"
  >
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <Fade>
      <div
        class="tile is-ancestor"
      >
        <div
          class="tile is-vertical"
        >
          <div
            class="tile"
          >
            <ServiceTile
              color="primary"
              icon="M5,1C5,3.7 6.56,6.16 9,7.32V22H11V15H13V22H15V7.31C17.44,6.16 19,3.7 19,1H17A5,5 0 0,1 12,6A5,5 0 0,1 7,1M12,1C10.89,1 10,1.89 10,3C10,4.11 10.89,5 12,5C13.11,5 14,4.11 14,3C14,1.89 13.11,1 12,1Z"
            />
            <ServiceTile
              color="warning"
              icon="M10,12A2,2 0 0,0 8,14A2,2 0 0,0 10,16A2,2 0 0,0 12,14A2,2 0 0,0 10,12M6,8A2,2 0 0,0 4,10A2,2 0 0,0 6,12A2,2 0 0,0 8,10A2,2 0 0,0 6,8M6,16A2,2 0 0,0 4,18A2,2 0 0,0 6,20A2,2 0 0,0 8,18A2,2 0 0,0 6,16M18,8A2,2 0 0,0 20,6A2,2 0 0,0 18,4A2,2 0 0,0 16,6A2,2 0 0,0 18,8M14,16A2,2 0 0,0 12,18A2,2 0 0,0 14,20A2,2 0 0,0 16,18A2,2 0 0,0 14,16M18,12A2,2 0 0,0 16,14A2,2 0 0,0 18,16A2,2 0 0,0 20,14A2,2 0 0,0 18,12M14,8A2,2 0 0,0 12,10A2,2 0 0,0 14,12A2,2 0 0,0 16,10A2,2 0 0,0 14,8M10,4A2,2 0 0,0 8,6A2,2 0 0,0 10,8A2,2 0 0,0 12,6A2,2 0 0,0 10,4Z"
            />
            <ServiceTile
              color="success"
              icon="M13.5,5.5C14.59,5.5 15.5,4.58 15.5,3.5C15.5,2.38 14.59,1.5 13.5,1.5C12.39,1.5 11.5,2.38 11.5,3.5C11.5,4.58 12.39,5.5 13.5,5.5M9.89,19.38L10.89,15L13,17V23H15V15.5L12.89,13.5L13.5,10.5C14.79,12 16.79,13 19,13V11C17.09,11 15.5,10 14.69,8.58L13.69,7C13.29,6.38 12.69,6 12,6C11.69,6 11.5,6.08 11.19,6.08L6,8.28V13H8V9.58L9.79,8.88L8.19,17L3.29,16L2.89,18L9.89,19.38Z"
            />
          </div>
        </div>
      </div>
    </Fade>
    <br />
    <br />
    <Fade>
      <ScheduleAppointment />
    </Fade>
  </div>
</body>
`;

exports[`Landing should render correctly LandingMain component 1`] = `
<section
  class="animated fadeIn hero is-info is-fullheight"
>
  <div
    class="hero-body"
  >
    <div
      class="column is-6 is-offset-3 has-text-centered"
    >
      <div
        class="tile is-parent"
      >
        <article
          class="tile is-child notification is-success landing"
        >
          <div
            class="content"
          >
            <p
              class="title"
            >
              Healthihost
            </p>
            <p
              class="subtitle"
            >
              Enabling the Vitality of the body Naturally
            </p>
          </div>
        </article>
      </div>
    </div>
  </div>
</section>
`;

exports[`Landing should render correctly LandingMain component 2`] = `
<section
  class="animated fadeIn hero is-info is-fullheight"
>
  <div
    class="hero-body"
  >
    <div
      class="column is-6 is-offset-3 has-text-centered"
    >
      <div
        class="tile is-parent"
      >
        <article
          class="tile is-child notification is-success landing"
        >
          <div
            class="content"
          >
            <p
              class="title"
            >
              Healthihost
            </p>
            <p
              class="subtitle"
            >
              Enabling the Vitality of the body Naturally
            </p>
          </div>
        </article>
      </div>
    </div>
  </div>
</section>
`;

exports[`Landing should render correctly ScheduleAppointment component 1`] = `
<div
  class="container has-text-centered"
>
  <div
    class="subtitle has-text-centered"
  >
    A healthy body allows us to live an optimal life with energy and enthusiasm. Do you want to be in balance, free from pain, able to be active, and doing your heartâ€™s desires?
  </div>
  <br />
  <div
    class="subtitle has-text-centered"
  >
    <NavBarLink
      exact={true}
      to="/appointment"
    >
      <div
        class="tile is-parent is-vertical grow more"
      >
        <article
          class="tile is-child notification has-text-white is-info"
        >
          <p
            class="title"
          >
            Schedule Appointment
          </p>
        </article>
      </div>
    </NavBarLink>
  </div>
</div>
`;

exports[`Landing should render correctly ServiceTile component 1`] = `
<NavBarLink
  exact={true}
  to="/massage"
>
  <article
    class="tile is-child notification has-text-white is-undefined grow"
  >
    <span
      class="icon is-large is-pulled-left"
    >
      <Icon
        color="white"
        horizontal={false}
        rotate={0}
        size={2}
        spin={false}
        vertical={false}
      />
    </span>
    <p
      class="title"
    >
      Massage Therapy
    </p>
    <br />
    <p
      class="subtitle"
    >
      Learn More
    </p>
  </article>
</NavBarLink>
`;
